src/bucket/HotArchiveBucketIndex.h

```diff
@@ -74,6 +74,13 @@ class HotArchiveBucketIndex : public NonMovableOrCopyable
         return mDiskIndex.scan(mDiskIndex.begin(), k).first;
     }
 
+    // Hot Archive does not support the cache, so define empty function for
+    // consistency with LiveBucketIndex
+    void
+    maybeAddToCache(std::shared_ptr<HotArchiveBucketEntry const> entry) const
```

Comment:
nit: this might be a micro-optimization, but since we don't always put an entry into a cache, we end up creating unnecessary shared ptr copies that are somewhat expensive. I don't see any significant downside to passing a const reference instead.